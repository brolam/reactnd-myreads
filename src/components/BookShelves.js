import React from 'react';
import PropTypes from 'prop-types'
import BookShelf from './BookShelf.js';

function BookShelves(props) {
    const { getBooksOnTheShelf, onChangeBookShelf, goToSearch } = props
    return (
        <div className="list-books">
            <div className="list-books-title">
                <h1>MyReads</h1>
            </div>
            <div className="list-books-content">
                <div>
                    <BookShelf
                        shelfTitle="Currently Readin"
                        books={getBooksOnTheShelf('currentlyReading')}
                        onChangeBookShelf={onChangeBookShelf}
                    />
                    <BookShelf
                        shelfTitle="Want to Read"
                        books={getBooksOnTheShelf('wantToRead')}
                        onChangeBookShelf={onChangeBookShelf}
                    />
                    <BookShelf
                        shelfTitle="Read"
                        books={getBooksOnTheShelf('read')}
                        onChangeBookShelf={onChangeBookShelf}
                    />
                </div>
            </div>
            <div className="open-search">
                <a id="searchButton" onClick={goToSearch}>Add a book</a>
            </div>
        </div>
    )
}

BookShelves.propTypes = {
    getBooksOnTheShelf: PropTypes.func.isRequired,
    onChangeBookShelf: PropTypes.func.isRequired,
    goToSearch: PropTypes.func.isRequired,
}

export default BookShelves