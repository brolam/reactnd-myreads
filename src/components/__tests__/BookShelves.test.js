import React from 'react';
import BookShelves from '../BookShelves.js';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';


const onChangeBookShelf = (e) => {}
const getBooksOnTheShelf = (shelf) => { return books.filter((book) => (book.shelf === shelf))}

test('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
    <BookShelves  
    onChangeBookShelf={onChangeBookShelf}
    getBooksOnTheShelf={getBooksOnTheShelf}
    goToSearch={() => { }} />, div)
})

test('Last Snapshot', () => {
    const bookShelves = renderer.create(
        <BookShelves  
        onChangeBookShelf={onChangeBookShelf}
        getBooksOnTheShelf={getBooksOnTheShelf}
        goToSearch={() => { }} />
      );
      let tree = bookShelves.toJSON();
      expect(bookShelves).toMatchSnapshot();
})

const books = [{
    id: "nggnmAEACAAJ",
    title: "The Linux Command Line",
    authors: ["William E. Shotts", "Breno Marques"],
    shelf: "currentlyReading"
  },
  {
    id: "PGR2AwAAQBAJ",
    title: "To Kill a Mockingbird",
    authors: ["Harper Lee", "Breno Marques"],
    shelf: "wantToRead"
  },
  {
    id: "yDtCuFHXbAYC",
    title: "Ender's Game",
    authors: ["Orson Scott Card", "Breno Marques"],
    shelf: "read"
  }
]