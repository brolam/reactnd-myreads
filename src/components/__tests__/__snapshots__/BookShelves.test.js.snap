// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Last Snapshot 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1>
      MyReads
    </h1>
  </div>
  <div
    className="list-books-content"
  >
    <div>
      <div
        className="bookshelf"
      >
        <h2
          className="bookshelf-title"
        >
          Currently Readin
        </h2>
        <div
          className="bookshelf-books"
        >
          <ol
            className="books-grid"
          >
            <li>
              <div
                className="book"
              >
                <div
                  className="book-top"
                >
                  <div
                    className="book-cover"
                    style={
                      Object {
                        "backgroundImage": "url(\\"https://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1\\")",
                        "height": 190,
                        "width": 120,
                      }
                    }
                  />
                  <div
                    className="book-shelf-changer"
                  >
                    <select
                      id="nggnmAEACAAJ"
                      onChange={[Function]}
                      value="currentlyReading"
                    >
                      <option
                        disabled={true}
                        value=""
                      >
                        Move to...
                      </option>
                      <option
                        value="currentlyReading"
                      >
                        Currently Reading
                      </option>
                      <option
                        value="wantToRead"
                      >
                        Want to Read
                      </option>
                      <option
                        value="read"
                      >
                        Read
                      </option>
                      <option
                        value="none"
                      >
                        None
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="book-title"
                >
                  The Linux Command Line
                </div>
                <div
                  className="book-authors"
                >
                  William E. Shotts, Breno Marques
                </div>
              </div>
            </li>
          </ol>
        </div>
      </div>
      <div
        className="bookshelf"
      >
        <h2
          className="bookshelf-title"
        >
          Want to Read
        </h2>
        <div
          className="bookshelf-books"
        >
          <ol
            className="books-grid"
          >
            <li>
              <div
                className="book"
              >
                <div
                  className="book-top"
                >
                  <div
                    className="book-cover"
                    style={
                      Object {
                        "backgroundImage": "url(\\"https://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1\\")",
                        "height": 190,
                        "width": 120,
                      }
                    }
                  />
                  <div
                    className="book-shelf-changer"
                  >
                    <select
                      id="PGR2AwAAQBAJ"
                      onChange={[Function]}
                      value="wantToRead"
                    >
                      <option
                        disabled={true}
                        value=""
                      >
                        Move to...
                      </option>
                      <option
                        value="currentlyReading"
                      >
                        Currently Reading
                      </option>
                      <option
                        value="wantToRead"
                      >
                        Want to Read
                      </option>
                      <option
                        value="read"
                      >
                        Read
                      </option>
                      <option
                        value="none"
                      >
                        None
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="book-title"
                >
                  To Kill a Mockingbird
                </div>
                <div
                  className="book-authors"
                >
                  Harper Lee, Breno Marques
                </div>
              </div>
            </li>
          </ol>
        </div>
      </div>
      <div
        className="bookshelf"
      >
        <h2
          className="bookshelf-title"
        >
          Read
        </h2>
        <div
          className="bookshelf-books"
        >
          <ol
            className="books-grid"
          >
            <li>
              <div
                className="book"
              >
                <div
                  className="book-top"
                >
                  <div
                    className="book-cover"
                    style={
                      Object {
                        "backgroundImage": "url(\\"https://books.google.com/books/content?id=yDtCuFHXbAYC&printsec=frontcover&img=1&zoom=1\\")",
                        "height": 190,
                        "width": 120,
                      }
                    }
                  />
                  <div
                    className="book-shelf-changer"
                  >
                    <select
                      id="yDtCuFHXbAYC"
                      onChange={[Function]}
                      value="read"
                    >
                      <option
                        disabled={true}
                        value=""
                      >
                        Move to...
                      </option>
                      <option
                        value="currentlyReading"
                      >
                        Currently Reading
                      </option>
                      <option
                        value="wantToRead"
                      >
                        Want to Read
                      </option>
                      <option
                        value="read"
                      >
                        Read
                      </option>
                      <option
                        value="none"
                      >
                        None
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="book-title"
                >
                  Ender's Game
                </div>
                <div
                  className="book-authors"
                >
                  Orson Scott Card, Breno Marques
                </div>
              </div>
            </li>
          </ol>
        </div>
      </div>
    </div>
  </div>
  <div
    className="open-search"
  >
    <a
      id="searchButton"
      onClick={[Function]}
    >
      Add a book
    </a>
  </div>
</div>
`;
